util/file_reader_writer.cc

```diff
@@ -88,7 +88,7 @@ Status RandomAccessFileReader::Read(uint64_t offset, size_t n, Slice* result,
       buf.AllocateNewBuffer(read_size);
       while (buf.CurrentSize() < read_size) {
         size_t allowed;
-        if (rate_limiter_ != nullptr) {
+        if (for_compaction_ && rate_limiter_ != nullptr) {
```

Comment:
this is a feature interaction bug fix for compaction read rate limiter combined with direct IO reads

Reply:
I included it here because it's necessary to make tests pass after this change.